projects/core/src/product/store/selectors/product.selectors.ts

```diff
@@ -33,46 +34,56 @@ export const getSelectedProductsFactory = (
 };
 
 export const getSelectedProductStateFactory = (
-  code: string
+  code: string,
+  scope?: string
 ): MemoizedSelector<StateWithProduct, LoaderState<Product>> => {
   return createSelector(
     getProductState,
-    details => StateEntityLoaderSelectors.entityStateSelector(details, code)
+    details =>
+      scope
+        ? StateEntityLoaderSelectors.entityStateSelector(details, code)[
+            scope
+          ] || initialLoaderState
```

Comment:
Shouldn't we put `|| initialLoaderState` after whole ternary operator?

Reply:
nope, because it's covered inside `entityStateSelector`, so applies to cases where there is no such a scope.